hpx/runtime/components/client_base.hpp

```diff
@@ -289,11 +289,18 @@ namespace hpx { namespace components
         }
 
         ///////////////////////////////////////////////////////////////////////
+#if defined(HPX_HAVE_COMPONENT_GET_GID_COMPATIBILITY)
         naming::id_type const & get_gid() const
         {
             return gid_.get();
         }
 
+#endif
+        naming::id_type const & get_id() const
+        {
+            return gid_.get();
+        }
+
```

Comment:
The usecase is, for example, a client that the user guarantees to keep alive, for that, he can use an unmanaged id for async calls. However, the real problem is if the client gets constructed from an unmanaged id to begin with, then this whole debate is meaningless after all.